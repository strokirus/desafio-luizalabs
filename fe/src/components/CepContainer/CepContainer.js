import React from 'react';
import { object, bool, func, number } from 'prop-types';
import { Map, TileLayer, Marker } from 'react-leaflet';

const CepContainer = ({ result, isLoading, error, closeCep }) => (
  <article>
    {isLoading &&
      <h3>Carregando</h3>
    }

    {error === 417 &&
      <h3>CEP inexistente</h3>
    }

    {error === 412 &&
      <h3>CEP mal formado</h3>
    }

    {result.logradouro &&
      <h1>{result.logradouro}</h1>
    }
    {result.complemento && result.complemento.length &&
      <p>{result.complemento}</p>
    }
    {result.bairro && result.bairro.length &&
      <p>{result.bairro}</p>
    }
    {result.localidade && result.localidade.length &&
      <p>{result.localidade} - {result.uf}</p>
    }
    {result.cep && result.cep.length &&
      <p>{result.cep}</p>
    }
    {closeCep &&
      <button onClick={closeCep}>x</button>
    }
    {result.geo !== undefined &&
      <Map center={[result.geo.lat, result.geo.lng]} zoom={16}>
        <TileLayer
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.osm.org/{z}/{x}/{y}.png"
        />
        <Marker position={[result.geo.lat, result.geo.lng]} />
      </Map>
    }
  </article>
);

CepContainer.propTypes = {
  result: object,
  isLoading: bool,
  error: number,
  closeCep: func,
};

CepContainer.defaultProps = {
  result: {},
  isLoading: true,
  error: 200,
  closeCep: () => {},
};

export default CepContainer;
