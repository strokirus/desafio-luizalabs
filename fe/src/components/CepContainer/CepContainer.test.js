import React from 'react';
import renderer from 'react-test-renderer';
import CepContainer from './CepContainer';

const props = {
  closeCep: jest.fn(),
  error: 200,
  result: {
    cep: '69028-401',
    logradouro: 'Rua Sargento Aguiar',
    complemento: 'PrÃ³ximo',
    bairro: 'Flores',
    localidade: 'Manaus',
    uf: 'AM',
    unidade: '',
    ibge: '1302603',
    gia: '',
    geo: undefined,
  },
  isLoading: false,
};

const propsError = {
  closeCep: jest.fn(),
  error: 417,
  result: undefined,
  isLoading: false,
};

describe('(Component) CepContainer', () => {
  it('should render an CepContainer with loading state', () => {
    const tree = renderer.create(<CepContainer isLoading closeCep={props.closeCep} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an CepContainer with result', () => {
    const tree = renderer.create(<CepContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an CepContainer with error 417', () => {
    const tree = renderer.create(<CepContainer {...propsError} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an CepContainer with error 412', () => {
    propsError.error = 412;
    const tree = renderer.create(<CepContainer {...propsError} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
