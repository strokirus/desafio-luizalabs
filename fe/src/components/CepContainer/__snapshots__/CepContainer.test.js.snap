// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) CepContainer should render an CepContainer with error 412 1`] = `
<article>
  <h3>
    CEP mal formado
  </h3>
  <button
    onClick={[MockFunction]}
  >
    x
  </button>
</article>
`;

exports[`(Component) CepContainer should render an CepContainer with error 417 1`] = `
<article>
  <h3>
    CEP inexistente
  </h3>
  <button
    onClick={[MockFunction]}
  >
    x
  </button>
</article>
`;

exports[`(Component) CepContainer should render an CepContainer with loading state 1`] = `
<article>
  <h3>
    Carregando
  </h3>
  <button
    onClick={[MockFunction]}
  >
    x
  </button>
</article>
`;

exports[`(Component) CepContainer should render an CepContainer with result 1`] = `
<article>
  <h1>
    Rua Sargento Aguiar
  </h1>
  <p>
    Pr√≥ximo
  </p>
  <p>
    Flores
  </p>
  <p>
    Manaus
     - 
    AM
  </p>
  <p>
    69028-401
  </p>
  <button
    onClick={[MockFunction]}
  >
    x
  </button>
</article>
`;
